File: iosApp/iosApp/Presentation/Main/MyInfo/MyInfoUiState.swift
//
//  MyInfoUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/06.
//

import Foundation
import Shared

struct MyInfoUiState {
    var isLoading: Bool
    var errorMessage: String?
    var isLoggedIn: Bool
    var user: User?
    var summary: MyPageSummary?
    var castDetail: CastDetail?
    var ownedCafes: [CafeManagementData.OwnedCafeSummary]
    var badges: [ProfileBadge]
    var popularCafes: [Cafe]
    var recentVisits: [Cafe]
    var favorites: [Cafe]
    var followedMaids: [Cast]
    var isLoginPromptVisible: Bool

    static let empty = MyInfoUiState(
        isLoading: false,
        errorMessage: nil,
        isLoggedIn: false,
        user: nil,
        summary: nil,
        castDetail: nil,
        ownedCafes: [],
        badges: [],
        popularCafes: [],
        recentVisits: [],
        favorites: [],
        followedMaids: [],
        isLoginPromptVisible: false
    )
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/myinfo/MyInfoUiState.kt
package com.hhp227.concafe.presentation.main.myinfo

import com.hhp227.concafe.domain.model.Cafe
import com.hhp227.concafe.domain.model.CafeManagementData
import com.hhp227.concafe.domain.model.Cast
import com.hhp227.concafe.domain.model.CastDetail
import com.hhp227.concafe.domain.model.MyPageSummary
import com.hhp227.concafe.domain.model.ProfileBadge
import com.hhp227.concafe.domain.model.User

data class MyInfoUiState(
    val isLoading: Boolean = false,
    val errorMessage: String? = null,
    val isLoggedIn: Boolean = false,
    val user: User? = null,
    val summary: MyPageSummary? = null,
    val castDetail: CastDetail? = null,
    val ownedCafes: List<CafeManagementData.OwnedCafeSummary> = emptyList(),
    val badges: List<ProfileBadge> = emptyList(),
    val popularCafes: List<Cafe> = emptyList(),
    val recentVisits: List<Cafe> = emptyList(),
    val favorites: List<Cafe> = emptyList(),
    val followedMaids: List<Cast> = emptyList(),
    val isLoginPromptVisible: Boolean = false
) {
    companion object {
        fun empty() = MyInfoUiState()
    }
}
